main.go

```diff
@@ -2,208 +2,243 @@ package main
+				set = setCmd - cas
 			} else {
-				e.bytes.WithLabelValues(dir).Set(float64(m))
+				log.Errorf("Failed to parse cas: %s", casErr)
 			}
+		} else {
+			log.Errorf("Failed to parse set %q: %s", s["cmd_set"], err)
 		}
+		ch <- prometheus.MustNewConstMetric(e.commands, prometheus.CounterValue, set, "set", "hit")
 
-		for _, st := range removalsStatuses {
-			m, err := strconv.ParseUint(stats[server][fmt.Sprintf("%s_unfetched", st)], 10, 64)
-			if err != nil {
-				e.removals.WithLabelValues(st, "unfetched").Set(0)
-			} else {
-				e.removals.WithLabelValues(st, "unfetched").Set(float64(m))
-			}
-		}
-		m, err = strconv.ParseUint(stats[server]["evicted"], 10, 64)
+		ch <- prometheus.MustNewConstMetric(e.currentBytes, prometheus.GaugeValue, parse(s, "bytes"))
+		ch <- prometheus.MustNewConstMetric(e.limitBytes, prometheus.GaugeValue, parse(s, "limit_maxbytes"))
+		ch <- prometheus.MustNewConstMetric(e.items, prometheus.GaugeValue, parse(s, "curr_items"))
+		ch <- prometheus.MustNewConstMetric(e.itemsTotal, prometheus.CounterValue, parse(s, "total_items"))
+
+		ch <- prometheus.MustNewConstMetric(e.bytesRead, prometheus.CounterValue, parse(s, "bytes_read"))
+		ch <- prometheus.MustNewConstMetric(e.bytesWritten, prometheus.CounterValue, parse(s, "bytes_written"))
+
+		ch <- prometheus.MustNewConstMetric(e.connections, prometheus.GaugeValue, parse(s, "curr_connections"))
+		ch <- prometheus.MustNewConstMetric(e.connectionsTotal, prometheus.CounterValue, parse(s, "total_connections"))
+
+		ch <- prometheus.MustNewConstMetric(e.evictions, prometheus.CounterValue, parse(s, "evictions"))
+		ch <- prometheus.MustNewConstMetric(e.reclaimed, prometheus.CounterValue, parse(s, "reclaimed"))
+	}
+
+}
+
+func parse(stats map[string]string, key string) float64 {
+	v, err := strconv.ParseFloat(stats[key], 64)
+	if err != nil {
+		log.Errorf("Failed to parse %s %q: %s", key, stats[key], err)
+		v = 0
+	}
+	return v
+}
+
+func sum(stats map[string]string, keys ...string) (float64, error) {
+	s := 0.
+	for _, key := range keys {
+		v, err := strconv.ParseFloat(stats[key], 64)
 		if err != nil {
-			e.removals.WithLabelValues("evicted", "fetched").Set(0)
-		} else {
-			e.removals.WithLabelValues("evicted", "fetched").Set(float64(m))
+			return 0, err
 		}
+		s += v
 	}
-
-	ch <- e.up
-	ch <- e.uptime
-	e.cache.Collect(ch)
-	e.usage.Collect(ch)
-	e.bytes.Collect(ch)
-	e.removals.Collect(ch)
+	return s, nil
 }
 
 func main() {
 	var (
+		timeout       = flag.Duration("memcached.timeout", time.Second, "memcached connect timeout.")
 		listenAddress = flag.String("web.listen-address", ":9106", "Address to listen on for web interface and telemetry.")
 		metricsPath   = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
 	)
-
 	flag.Parse()
 
-	for _, server := range flag.Args() {
-		exporter := NewExporter(server)
-		prometheus.MustRegister(exporter)
+	servers := flag.Args()
```

Comment:
This is actually not a best practice in prometheus. Exporters are supposed to be deployed right next to each process and only scrape one each. I guess we should kill that and the `server` label and use a `--memcached.address` flag instead. wdyt @brian-brazil?